rust/relay/src/proptest.rs

```diff
@@ -31,9 +30,10 @@ pub fn channel_number() -> impl Strategy<Value = ChannelNumber> {
 }
 
 pub fn channel_payload() -> impl Strategy<Value = (Vec<u8>, usize)> {
-    vec(any::<u8>(), 0..(u16::MAX as usize)).prop_flat_map(|vec| {
+    any::<Vec<u8>>().prop_map(|vec| {
         let len = vec.len();
-        (Just(vec), 0..len)
+
+        (vec, len)
     })
```

Comment:
So instead of saying "Generate a length from 0 to `u16::MAX - 1` and then make a `Vec<u8>` of that length", you're just directly saying, "Generate a `Vec<u8>`"?

And that avoids the problem of saying, "Generate a Vec with a length that's at least 0, but less than 0"?

That part makes sense.

But without mentioning `u16`, does this change the possible sizes? Can it try to generate a Vec of length `usize::MAX - 1`?

LGTM as long as it won't cause out of memory errors.

Reply:
Hmm, now that I think of it, this may blow up if we exceed the max supported length of a channel data packet.